modules/social_features/social_profile/modules/social_profile_privacy/social_profile_privacy.module

```diff
@@ -362,3 +410,17 @@ function _social_profile_privacy_resave_search_indexes() {
     }
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ *
+ * Set the correct profile name.
+ */
+function social_profile_privacy_profile_presave(EntityInterface $profile) {
+  /** @var \Drupal\social_profile_privacy\Service\SocialProfilePrivacyBatchHelper $batch_helper */
+  $batch_helper = \Drupal::service('social_profile_privacy.batch.helper');
+  if ($profile instanceof ProfileInterface) {
```

Comment:
It would be great if we add few more checks before directly updating the profile. You can imagine that a user can update profile values other than names.

Reply:
This change has led to the change of a lot of the created functionality. But I did it and I think it looks better😀 Thanks for the suggestions. You can check how it looks now.